library/StateControl.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import json
+
+from library.common import *
+from library.ImageControl import ImageControl
+from library.PreseedControl import PreseedControl
+
+class StateControl(object):
+    def __init__(self, requester, machine_id):
+        self.requester = requester
+        self.machine_id = machine_id
+
+    def machine_provision(self):
+        """ enables netboot on the machine and pxe boots it """
+        url = "/api/v1/machine/{}/state".format(self.machine_id)
+
+        data = json.dumps({'state': 'provision'})
+
+        try:
+            return self.requester.post(url, data)
+        except Exception as err:
+            raise err
+
+    def set_machine_parameters(self, preseed_name=None, initrd_desc=None,
+                               kernel_desc=None, kernel_opts="",
+                               arch=None, subarch=None):
+        """ Set parameters on machine specified by machine_id """
+        url = "/api/v1/machine/{}".format(self.machine_id)
+
+        image_controller = ImageControl(self.requester)
+        preseed_controller = PreseedControl(self.requester,preseed_name)
```

Comment:
Can the preseed be null for PXE boot? If so, won't this throw an error?

Reply:
Yes, according to the [API](https://github.com/mr-provisioner/mr-provisioner/blob/f6805794ac672940a0c1849c9a547dc789f71ee3/mr_provisioner/api/v1/controllers.py#L623) only kernel, subarch and the associated bootloader (as well as a BMC) are needed for provisioning (i.e. PXE (re)booting)

Reply:
In this implementation, will it fail silently or throw an exception?

Reply:
If the HTTP return code is not normal for REST requests, the implementation (Requester) throws an exception. (so != 200 for GETs and PUTs and DELETEs and != 201 for POSTs)